scripts/gen_app_partitions.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+import sys
+import argparse
+import os
+import re
+import string
+
+
+# This script will create linker comands for power of two aligned MPU
+# when APP_SHARED_MEM is enabled.
+print_template = """
+/* Auto generated code do not modify */
+. = ALIGN( 1 << LOG2CEIL(data_smem_REPLACE_STRINGb_end - data_smem_REPLACE_STRING));
+data_smem_REPLACE_STRING = .;
+KEEP(*(SORT(data_smem_REPLACE_STRING*)))
+. = ALIGN(_app_data_align);
+data_smem_REPLACE_STRINGb_end = .;
+. = ALIGN( 1 << LOG2CEIL(data_smem_REPLACE_STRINGb_end - data_smem_REPLACE_STRING));
+"""
+
+
+def find_partitions(filename, fileOutput):
+    list_with_declarations =[]
+    string = ''
+    with open(filename, 'r') as f:
+        lines = f.read()
+        list_with_declarations.append(re.findall(r'PARTITIONS_FOR_EACH_CREATE(.*)', lines))
+        if list_with_declarations == []:
+            return
+    for each_declaration in list_with_declarations:
+        partitions = each_declaration[0].replace(");","").replace("(","").replace(" ", "").split(",")
```

Comment:
Yikes.    This is... outrageously fragile.  You're demanding that every partition defined by done with exactly the right macro (which isn't even named "DEFINE" or "DECLARE" or anything), that the whitespace used to invoke it be compatible with your magic string replacements here that try to turn the C preprocessor syntax into valid python, and that the symbols passed to the macro not themselves be subject to preprocessor hackery (i.e. they have to be literals).

No.  That's too ugly.

You can use the preprocessor to declare some data for inspection later using a custom data section.   We do that in other situations already.  Please investigate a trick like that.

Reply:
>You're demanding that every partition defined by done with exactly the right macro 

yes that's how it was before as well. How is it different now??

>and that the symbols passed to the macro not themselves be subject to preprocessor hackery (i.e. they have to be literals).

Agreed it has to be literals. Else this wont work.  But if they aren't literals I cant make it to work.
The 2 power alignment issue prohibits me from manipulating any alignment after the prebuilt is ready.
I found this is the best way to make this feature work. 
Else all I can see is revert the faulty PR.

> You can use the preprocessor to declare some data for inspection later using a custom data section. We do that in other situations already.

Please provide instances where the data is accessible before prebuilt is ready.